### PR TITLE
Feature: Query reccomendations

### PR DESCRIPTION
**Pull request by gikoifma**
### Summary

Add Query Recommendations (Suggestions + Warnings).

 ### Other Information

Add configurations to enable/disable recommendations by severity or by specific code (Each diagnostic has a unique code as an identifier).

Since in KustoLanguageFeatures there were formatting changes, I'll mention the functions in which there was an actual change (for easier review process):
- getCommandAndLocationInContext
- doValidation
- toLsDiagnostics

### CODE REVIEW

File: package/src/languageService/kustoLanguageService.ts [hunk 1]
```diff
@@ -674,12 +674,21 @@ class KustoLanguageService implements LanguageService {
...
-                    return diagnostics;
+                let diagnostics = this.toArray<Kusto.Language.Diagnostic>(block.Service.GetDiagnostics(false));
+                const enableWarnings = includeWarnings ?? this._languageSettings.enableQueryWarnings;
+                const enableSuggestions = includeSuggestions ?? this._languageSettings.enableQuerySuggestions;
+                if (enableWarnings || enableSuggestions) {
+                    // Concat Warnings and suggestions to the diagnostics
+                    const analyzerDiagnostics = block.Service.GetAnalyzerDiagnostics(null, true);
+                    const filterredDiagnostics = this.toArray<Kusto.Language.Diagnostic>(analyzerDiagnostics).filter(d => {
+                        const allowSeverity = (enableWarnings && d.Severity === 'Warning') || (enableSuggestions && d.Severity === 'Suggestion');
+                        const allowCode = !this._languageSettings.disabledDiagnoticCodes?.includes(d.Code);
```
Comment (izikl) [inline package/src/languageService/kustoLanguageService.ts:685]: What are diagnostoc codes?

Comment (gikoifma) [reply package/src/languageService/kustoLanguageService.ts:685]: Like an id

File: package/src/languageService/kustoLanguageService.ts [hunk 2]
```diff
@@ -674,12 +674,21 @@ class KustoLanguageService implements LanguageService {
 
         const diagnostics = blocks
             .map((block) => {
-                const diagnostics = this.toArray<Kusto.Language.Diagnostic>(block.Service.GetDiagnostics());
-                if (diagnostics) {
-                    return diagnostics;
+                let diagnostics = this.toArray<Kusto.Language.Diagnostic>(block.Service.GetDiagnostics(false));
```
Comment (izikl) [inline package/src/languageService/kustoLanguageService.ts:677]: lets rename diagnostics to errorsDiagnostics 
and analyzerDiagnostics to warningAndSuggestionDiagnostics

**Review by lidermanrony — DISMISSED** [review]

**Comment by lidermanrony** [comment]
don't forget to advance package version otherwise npm publish will not work

**Review by izikl — DISMISSED** [review]

File: package/src/languageService/kustoLanguageService.ts [hunk 3]
```diff
@@ -660,12 +660,23 @@ class KustoLanguageService implements LanguageService {
         return Promise.resolve(newDatabasesReferences);
     }
 
+    getPossibleAnalyzerDiagnostics() {
+        const analyzers: k2.KustoAnalyzer[] = (Bridge as any).toArray(k2.KustoAnalyzers.All);
```
[the remark this answers is not in the document]
Comment (gikoifma) [reply package/src/languageService/kustoLanguageService.ts]: removed

**Review by lidermanrony — APPROVED** [review]

**Pull request merged**